### PR TITLE
[A3] Add section on open-source LLMs with few more links and resources

### PR DESCRIPTION
**Pull request by rish-16**
Hey! I've added a few more resources and links, specifically focusing on the open-source LLM offerings and how to finetune such models. The attached guides are very useful and have made waves in my Twitter circles, they are pretty reliable IMO.

Do let me know if there are any changes to be made :)

### CODE REVIEW

File: website/pages/coursework/artificial-intelligence.mdx [hunk 1]
```diff
@@ -529,6 +531,26 @@ The following platforms improve the experience of using LLMs in production. They
...
 
+#### Open-source LLMs
+
+Platforms like HuggingFace offer support for popular open-source language models born out of community efforts to recreate existing proprietary LLMs provided by miscellaneous companies via APIs or software. These open-source LLMs are easily [finetunable](https://cs3216.com/coursework/artificial-intelligence#fine-tuning) and 
+lightweight. While not as powerful as commercially available LLMs, they offer a free, discounted, yet surprisingly expressive experience in terms of text generation and reasoning capabilities. Here are a few popular models and associated guides to finetune them:
+
+- LLAMA 2 ([`code`](), [`finetuning guide`](https://towardsdatascience.com/fine-tune-your-own-llama-2-model-in-a-colab-notebook-df9823a04a32), [`paper`](https://scontent-xsp1-1.xx.fbcdn.net/v/t39.2365-6/10000000_662098952474184_2584067087619170692_n.pdf?_nc_cat=105&ccb=1-7&_nc_sid=3c67a6&_nc_ohc=bGkSwJy8Xa4AX9SX9XW&_nc_ht=scontent-xsp1-1.xx&oh=00_AfC2wTMNDOi-1j5BHP7STSfurMRMQQANleVSDe7vcMVBOQ&oe=64FDAEFF))
```
Comment (yangshun) [inline website/pages/coursework/artificial-intelligence.mdx]: Are the `code` links meant to be added later? Otherwise it's better to remove empty links

Comment (yangshun) [inline website/pages/coursework/artificial-intelligence.mdx]: This list probably works well as a table

**Review by yangshun — COMMENTED** [review]
Thank you @rish-16! Just a question about the code links, otherwise we're ready to merge.

File: website/pages/coursework/artificial-intelligence.mdx (hunk 1 above)
Comment (rish-16) [reply website/pages/coursework/artificial-intelligence.mdx]: Oh, oops, I changed my convention from code to finetuning guide. Think I forgot to remove the hanging ones with no link. Will remove!

Comment (rish-16) [reply website/pages/coursework/artificial-intelligence.mdx]: Can I please check if this .mdx format you’re using is essentially just fancy markdown? So if I want to create a table for these links, I can create an old-fashioned md table using the +-| symbols

**Comment by rish-16** [comment]
Different models have different resources available at the moment, so the ones I've mentioned don't have standardised labels (ie, `code`, `finetuning guide`, `paper`, etc)

**Review by yangshun — APPROVED** [review]
Looks good! Thanks a lot @rish-16

**Pull request merged**